ziggurat-core-geoip/Cargo.toml

```
@@ -11,9 +11,9 @@ description = "GeoIP localization package for Ziggurat"
 [dependencies]
 async-trait = "0.1.63"
 geoutils = "0.5.1"
-ip2location = "0.3.5"
+ip2location = "0.4.2"
```

A general file comment:

Let's not specify the patch version of the SemVer, no need for that.
And it will make maintenance easier
```suggestion
ip2location = "0.4"
```

---

I can't see right now any big advantage of such approach. 

What makes maintanance easier when specifying only major and minor?  As the doc states:
```
1.2.3  :=  >=1.2.3, <2.0.0
1.2    :=  >=1.2.0, <2.0.0
1      :=  >=1.0.0, <2.0.0
0.2.3  :=  >=0.2.3, <0.3.0
0.2    :=  >=0.2.0, <0.3.0
0.0.3  :=  >=0.0.3, <0.0.4
0.0    :=  >=0.0.0, <0.1.0
0      :=  >=0.0.0, <1.0.0
```

taking the newest package version rules are the same when we specify full semver or just two left-most numbers. So there is no worry that specyfing patch would cause any trouble with updates because `cargo` would choose newest version anyway. And updating package to any version newer than the automatic update can do demands the same work (changing the same line in Cargo.toml and tests).

So why I'm trying to specify full version in most cases? IMHO there is at least one advantage - safety. 
When specifying exact version (and newer ones) let's say that I promise my user I've tested and checked the code to be working correctly AT LEAST using that version.
What if previous version of the package contained a bug that would make my program crash? Maybe I'm lazy but I don't want to waste my time checking changelog for each package what've been changed between the PATCH versions and think if user can use package eg. 1.1.1 safely while there is available 1.1.10 version for that package. I don't even consider why someone would like to pick older version of some crate but I feel comfortable knowing that my program would work correctly and there is no point (without manual intervention - but then user take reponsibility 😄 ) to build it with any outdated package.
Moreover, sometimes there can be safety threats in older versions of the crates and why should we even allow anyone to work with such crates which have known security flaws? Just like we've here:
[https://deps.rs/repo/github/runziggurat/xrpl](https://deps.rs/repo/github/runziggurat/xrpl)
with the `chrono` crate. It's reported as `maybe insecure` because there are existing security threats in older versions and we're allowing to build up our product with those versions just because we do not demand `chrono` in some minimal (bug free) version.